ethereum/solidity/contracts/DataRootTuple.sol

```diff
@@ -0,0 +1,14 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.0;
+
+/// @notice A tuple of data root with metadata. Each data root is associated
+///  with a Celestia block height.
+/// @dev `availableDataRoot` in
+///  https://github.com/celestiaorg/celestia-specs/blob/master/src/specs/data_structures.md#header
+struct DataRootTuple {
+    // Celestia block height the data root was included in.
+    // Genesis block is height = 1.
+    uint256 height;
+    // Data root.
+    bytes32 dataRoot;
```

Comment:
Shouldn't this be a commitment over a set of data roots ? 
@evan-forbes In celestia-app, we're working on commitments over set of data roots ? Thus, what will arrive in the smart contract is just a commitment and range (begin_block -> end_block). Correct me please if I am wrong.

Reply:
A single data root tuple is for a single data root. The trick is that when you submit, you submit to `submitDataRootTupleRoot` (notice the _second_ root at the end). So each `DataRootTuple` is a leaf in a tree, and the root of that tree is what's posted.

Reply:
Aahh get it :+1: Then, maybe we should add a comment specifying this so that newcomers don't have the same misunderstanding.